CollectionViewShelfLayout/CollectionViewShelfLayout.swift

```diff
@@ -170,6 +170,10 @@ open class CollectionViewShelfLayout: UICollectionViewLayout {
         cellPanningScrollViews.append(panningScrollView)
       }
       
+      for oldPanningScrollView in oldPanningScrollViews {
+        oldPanningScrollView.removeFromSuperview()
```

Comment:
We should call `removeFromSuperview()` in the `willSet` `trackingView` property observer and also how's about calling `oldPanningScrollView.trackingView = nil` within a defer block after assigning the oldPanningScrollViews variable.